src/pages-and-resources/discussions/app-config-form/apps/legacy/LegacyConfigForm.jsx

```diff
@@ -67,19 +95,22 @@ function LegacyConfigForm({
           touched,
         },
       ) => {
-        const { discussionTopics } = values;
-        const discussionTopicErrors = discussionTopics.map((value, index) => Boolean(
-          touched.discussionTopics
-          && touched.discussionTopics[index]?.name
-          && errors.discussionTopics
-          && errors?.discussionTopics[index]?.name,
+        const { discussionTopics, blackoutDates } = values;
+        const discussionTopicErrors = discussionTopics.map((value, index) => (
+          checkFieldErrors(touched, errors, 'discussionTopics', 'name', index)
         ));
+        const blackoutDatesErrors = blackoutDates.map((value, index) => (
+          checkFieldErrors(touched, errors, 'blackoutDates', 'startDate', index)
+          || checkFieldErrors(touched, errors, 'blackoutDates', 'endDate', index)
+          || checkFieldErrors(touched, errors, 'blackoutDates', 'startTime', index)
+          || checkFieldErrors(touched, errors, 'blackoutDates', 'endTime', index)));
+
```

Comment:
Reminder--> I will be looking into the need of this once again. as I am confused why we need this when we have formik errors object

Reply:
self note: https://formik.org/docs/guides/validation

Reply:
suppose clicks on field#1 and leave the field empty. our formik validation runs onBlur of field#1 and validates the schema. So error object contains all errors including the non-touched fields. So we only need to display errors for touched fields. So here I am checking the error or touched state of the field.